src/ActionParameterHandler.ts

```diff
@@ -138,8 +138,41 @@ export class ActionParameterHandler {
                     value = this.transformValue(value, param);
                     value = this.validateValue(value, param); // note this one can return promise
                 }
+                return value;
+        }
+    }
+
+    /**
+     * Normalizes string value to number or boolean.
+     */
+    protected normalizeValue(value: any, type: string) {
+        switch (type) {
+            case "number":
+                if (value === "")
+                    return undefined;
+                const valueNumber = Number(value);
+                // tslint:disable-next-line:triple-equals
+                if (valueNumber == value)
+                    return valueNumber;
+                else
+                    throw new BadRequestError(`${value} can't be parsed to number.`);
+
+            case "string":
+                return value;
+
+            case "boolean":
+                if (value === "true" || value === "1") {
+                    return true;
+
+                } else if (value === "false" || value === "0") {
+                    return false;
+                }
+
+                return Boolean(value);
+
+            default:
```

Comment:
I think we should also cover the `date` case and also the array feature which was requested in #123 - it is possible with class-transform `@Type` decorator to work with arrays.